src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -220,13 +223,14 @@ def add_extension(cmd, source=None, extension_name=None, index_url=None, yes=Non
             logger.debug(err)
             raise CLIError("No matching extensions for '{}'. Use --debug for more information.".format(extension_name))
     extension_name = _add_whl_ext(cmd=cmd, source=source, ext_sha256=ext_sha256,
-                                  pip_extra_index_urls=pip_extra_index_urls, pip_proxy=pip_proxy)
-    _augment_telemetry_with_ext_info(extension_name)
+                                  pip_extra_index_urls=pip_extra_index_urls, pip_proxy=pip_proxy, system=system)
+    ext = get_extension(extension_name)
+    _augment_telemetry_with_ext_info(extension_name, ext)
     try:
-        if extension_name and get_extension(extension_name).experimental:
+        if extension_name and ext.experimental:
             logger.warning("The installed extension '%s' is experimental and not covered by customer support. "
                            "Please use with discretion.", extension_name)
-        elif extension_name and get_extension(extension_name).preview:
+        elif extension_name and ext.preview:
             logger.warning("The installed extension '%s' is in preview.", extension_name)
```

Comment:
get_extension calls get_extensions to load all extension, then fetch the extension with the name, we should avoid calling it mulitple times.